services/src/main/java/org/apache/druid/cli/CliPeon.java

```diff
@@ -144,10 +143,10 @@ public class CliPeon extends GuiceRunnable
 {
   @SuppressWarnings("WeakerAccess")
   @Required
-  @Arguments(description = "baseTaskDirPath taskId attemptId")
+  @Arguments(description = "taskDirPath attemptId")
   public List<String> taskAndStatusFile;
 
-  // path to the base task Directory
+  // path to the task Directory
   private String taskDirPath;
```

Comment:
I have some difficulty understanding the connection of this path with the intermediate directories being created by the task. This path would have been derived from the multiple directory selection algorithm present in the task runners. Although, `CliPeon` also injects `TaskConfig` object as a config object which means that config object wouldn't have the directory selection logic present. 
Is there some place which modifies the `TaskConfig` object to override its `baseTaskDirPath` with the path given in the task launcher command line?

Reply:
Yeah, you caught a bug where I wasn't properly overriding this when forking processes.  That's fixed now.